CHANGELOG/README.md

```diff
@@ -0,0 +1,98 @@
+# CHANGELOG
+
+The CHANGELOG directory is used for individual changelog file practice.
+The `kong/CHANGELOG.md` now is deprecated.
+
+
+## How to add a changelog file for your PR?
+
+1/ Copy the `changelog-template.yaml` file and rename with your PR number or a short message as the filename. For example, `11279.yaml`, `introduce-a-new-changelog-system.yaml`. (Prefer using PR number as it's already unique and wouldn't introduce conflict)
+
+2/ Fill out the changelog template.
+
+
+The description of the changelog file field, please follow the `schema.json` for more details.
+
+- message: Message of the changelog
+- type: Changelog type
+- scope: Changelog scope
+- prs: List of associated GitHub PRs
+- issues: List of associated GitHub issues
+- jiras: List of associated Jira tickets for internal track
+
+Sample 1
+```yaml
+message: Introduce the request id as core feature.
+type: feat
+scope: Core
+prs:
+  - 11308
+```
+
+Sample 2
+```yaml
+message: Fix response body gets repeated when `kong.response.get_raw_body()` is called multiple times in a request lifecycle.
+type: bugfix
+scope: PDK
+prs:
+  - 11424
+jiras:
+  - "FTI-5296"
+```
+
+
+## changelog command
+
+The `changelog` command tool provides `add`, `preview`, and `release` commands.
+
+### Prerequisites
+
+You can skip this part if you're at Kong Bazel virtual env.
+
+Install luajit
+
+Install luarocks libraries
+
+```
+luarocks install penlight --local
+luarocks install lyaml --local
+```
+
+### Usage
+
+```shell
+$ ./changelog -h
+
+Usage: changelog <command> [options]
+
+Commands:
+  add <filename> [options]           add a changelog file.
+  release <version> [options]        release a release note based on the files in the CHANGELOG/unreleased directory.
+  preview <version> [options]        preview a release note based on the files in the CHANGELOG/unreleased directory.
+
+Options:
+  -h, --help                         display help for command
+  -m, --message (optional string)    changelog message
+  -t, --type (optional string)       changelog type
```

Comment:
`--type` and `--message` should not be optional.  If you don't want to prompt the user, make it an error to not provide these options.  It would then be better to change the signature to not require the options, i.e. `changelog add <type> <message>`.

Reply:
Setting `type` and `message` to optional does not mean they're not required, it means they're not required while creating a changelog file.

Reply:
Please make the tool more user friendly.  If it is hard to use, developers will waste time figuring out what to do.  It should be easy to provide a tool that prevents developers from making mistakes in the first place, so please make that your goal.